pcsx2/GS/Renderers/SW/GSDrawScanline.cpp

```diff
@@ -1291,8 +1291,11 @@ __ri void GSDrawScanline::CDrawScanline(int pixels, int left, int top, const GSV
 				GSVector4i fga = global.fga;
 #endif
 
-				rb = frb.lerp16<0>(rb, fog);
-				ga = fga.lerp16<0>(ga, fog).mix16(ga);
+				// Use non-rounding interpolation for fog (PS2 hardware doesn't round)
+				// Manual implementation of lerp16<0> without rounding
+				// rb = frb + (rb - frb) * fog / 32768 (equivalent to lerp16<0> but without rounding)
+				rb = frb.add16(rb.sub16(frb).sll16<1>().mul16hs(fog));
+				ga = fga.add16(ga.sub16(fga).sll16<1>().mul16hs(fog)).mix16(ga);
```

Comment:
The C implementation is supposed to be a reference for the asm implementations, they should be fixed the same way